regulation-worker/internal/delete/api/api.go

```diff
@@ -49,11 +53,39 @@ func (api *APIManager) Delete(ctx context.Context, job model.Job, destConfig map
 	}
 	req.Header.Set("Content-Type", "application/json")
 
+	var tokenStatusCode int
+	var accountSecretInfo *oauth.AuthResponse
+	// identifier to know if the destination supports OAuth for regulation-api
+	rudderUserDeleteAccountId, delAccountIdExists := destDetail.Config["rudderDeleteAccountId"]
+	if delAccountIdExists {
+		// Fetch Token call
+		// Get Access Token Information to send it as part of the event
+		tokenStatusCode, accountSecretInfo = api.OAuth.FetchToken(&oauth.RefreshTokenParams{
+			AccountId:       rudderUserDeleteAccountId.(string),
+			WorkspaceId:     job.WorkspaceID,
+			DestDefName:     destDetail.Name,
+			EventNamePrefix: "fetch_token",
+		})
+		pkgLogger.Debugf(`[%s][FetchToken] Token Fetch Method finished (statusCode, value): (%v, %+v)`, destDetail.Name, tokenStatusCode, accountSecretInfo)
+		if tokenStatusCode != http.StatusOK {
+			pkgLogger.Errorf(`[%s][FetchToken] Error in Token Fetch statusCode: %d\t error: %s\n`, destDetail.Name, tokenStatusCode, accountSecretInfo.Err)
+		}
+		// setting oauth related information
+		payload, marshalErr := json.Marshal(accountSecretInfo.Account)
+		if marshalErr != nil {
+			pkgLogger.Errorf("error while marshalling account secret information: %v", marshalErr)
+			return model.JobStatusFailed
+		}
+		req.Header.Set("X-Rudder-Dest-Info", string(payload))
```

Comment:
would it make things more readable, if we move this complete `OAuth` fetching and marshalling thing to a separate function & set it there.
```suggestion
err:=api.setOAuthToken(req)
if err!=nil{
model.JobStatusFailed
}
```

Reply:
Slightly modified the implementation, please take a look